src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Commands/DependenciesContextMenuProvider.cs

```diff
@@ -0,0 +1,157 @@
+            }
+            else if (projectItem.Flags.Contains(DependencyTreeFlags.ComDependency))
+            {
+                menuCommandId = Menus.IDM_VS_CTXT_COMREFERENCE;
+            }
+            else if (projectItem.Flags.Contains(DependencyTreeFlags.ProjectSubTreeRootNode))
+            {
+                menuCommandId = Menus.IDM_VS_CTXT_PROJECTREFERENCE_GROUP;
+            }
+            else if (projectItem.Flags.Contains(DependencyTreeFlags.ProjectDependency))
+            {
+                menuCommandId = Menus.IDM_VS_CTXT_PROJECTREFERENCE;
+            }
+            else if (projectItem.Flags.Contains(DependencyTreeFlags.SharedProjectDependency))
+            {
+                menuCommandId = Menus.IDM_VS_CTXT_SHAREDPROJECTREFERENCE;
+            }
+            else if (projectItem.Flags.Contains(DependencyTreeFlags.AnalyzerSubTreeRootNode))
+            {
+                menuCommandId = Menus.IDM_VS_CTXT_ANALYZERREFERENCE_GROUP;
+            }
+            else if (projectItem.Flags.Contains(DependencyTreeFlags.AnalyzerDependency))
+            {
+                menuCommandId = Menus.IDM_VS_CTXT_ANALYZERREFERENCE;
+            }
+            else if (projectItem.Flags.Contains(DependencyTreeFlags.FrameworkSubTreeRootNode))
+            {
+                menuCommandId = Menus.IDM_VS_CTXT_FRAMEWORKREFERENCE_GROUP;
+            }
+            else if (projectItem.Flags.Contains(DependencyTreeFlags.FrameworkDependency))
+            {
+                menuCommandId = Menus.IDM_VS_CTXT_FRAMEWORKREFERENCE;
+            }
+            else if (projectItem.Flags.Contains(DependencyTreeFlags.SdkSubTreeRootNode))
+            {
+                menuCommandId = Menus.IDM_VS_CTXT_SDKREFERENCE_GROUP;
+            }
+            else if (projectItem.Flags.Contains(DependencyTreeFlags.SdkDependency))
+            {
+                menuCommandId = Menus.IDM_VS_CTXT_SDKREFERENCE;
+            }
+            else
+            {
+                menuCommandGuid = default;
+                menuCommandId = default;
+                return false;
+            }
+
+            menuCommandGuid = VSConstants.CMDSETID.ShellMainMenu_guid;
+            return true;
+        }
+
+        public bool TryGetMixedItemsContextMenu(IEnumerable<IProjectTree> projectItems, out Guid menuCommandGuid, out int menuCommandId)
+        {
+            Requires.NotNull(projectItems, nameof(projectItems));
+
+            menuCommandGuid = default;
+            menuCommandId = default;
+
+            // If there are multiple items, and any item is a group node, suppress the context menu altogether
+
+            int count = 0;
+            bool containsProhibited = false;
+
+            foreach (IProjectTree item in projectItems)
+            {
+                count++;
+
+                if (!containsProhibited)
+                {
+                    // TODO when CPS inserts, use ContainsAny here instead
```

Comment:
This needs explanation, I don't understand what its trying to prevent.

Reply:
The current code is fine. It could just be a little cleaner once we have newer CPS bits inserted.